src/components/Shared/DatePicker.vue

```diff
@@ -0,0 +1,228 @@
+  -
+  -->
+
+<template>
+	<DatetimePicker
+		:lang="lang"
+		:first-day-of-week="firstDay"
+		:format="format"
+		:value="date"
+		:type="type"
+		:clearable="false"
+		:minute-step="5"
+		:not-before="minimumDate"
+		:not-after="maximumDate"
+		@change="change">
+		<template
+			v-if="displayTimezoneIcon"
+			slot="calendar-icon">
+			<button
+				class="datetime-picker-inline-icon icon icon-timezone"
+				:class="{ 'datetime-picker-inline-icon--highlighted': highlightTimezone }"
+				@click.stop.prevent="toggleTimezonePopover" />
+			<Popover
+				:open.sync="showTimezonePopover"
+				open-class="timezone-popover-wrapper">
+				<div class="timezone-popover-wrapper__title">
+					<strong>
+						{{ $t('calendar', 'Please select a timezone:') }}
+					</strong>
+				</div>
+				<TimezoneSelect
+					class="timezone-popover-wrapper__timezone-select"
+					:value="timezoneId"
+					@change="changeTimezone" />
+			</Popover>
+		</template>
+	</DatetimePicker>
+</template>
+
+<script>
+import { DatetimePicker } from '@nextcloud/vue/dist/Components/DatetimePicker'
+import { Popover } from '@nextcloud/vue/dist/Components/Popover'
+import { getDayNamesShort, getMonthNamesShort, getFirstDay } from '@nextcloud/l10n'
+import moment from '@nextcloud/moment'
+
+import TimezoneSelect from './TimezoneSelect'
+import loadMomentLocalization from '../../utils/moment.js'
+
+export default {
+	name: 'DatePicker',
+	components: {
+		DatetimePicker,
+		Popover,
+		TimezoneSelect
+	},
+	props: {
+		date: {
+			type: Date,
+			required: true
+		},
+		hasTimezone: {
+			type: Boolean,
+			default: false
+		},
+		timezoneId: {
+			type: String,
+			default: null
+		},
+		prefix: {
+			type: String,
+			default: null
+		},
+		isAllDay: {
+			type: Boolean,
+			required: true
+		},
+		userTimezoneId: {
+			type: String,
+			default: null
+		},
+		min: {
+			type: Date,
+			default: null
+		},
+		max: {
+			type: Date,
+			default: null
+		}
+	},
+	data() {
+		return {
+			locale: 'en',
+			lang: {
+				days: getDayNamesShort(),
+				months: getMonthNamesShort(),
+				placeholder: {
+					date: t('calendar', 'Select Date')
+				}
+			},
+			firstDay: getFirstDay() === 0 ? 7 : getFirstDay(),
+			showTimezonePopover: false
+		}
+	},
+	computed: {
+		/**
+		 * Whether or not to display the timezone icon
+		 * and allow the user to pick a different timezone
+		 *
+		 * @returns {Boolean}
+		 */
+		displayTimezoneIcon() {
+			return !this.isAllDay && this.timezoneId !== null
+		},
+		/**
+		 * Whether or not to highlight the timezone-icon.
+		 * The icon is highlighted when the selected timezone
+		 * does not equal the current user's timezone
+		 *
+		 * @returns {Boolean}
+		 */
+		highlightTimezone() {
+			return this.timezoneId !== this.userTimezoneId
+		},
+		/**
+		 * Formats the date string
+		 *
+		 * @returns {String}
+		 */
+		format() {
+			const parts = []
+
+			if (this.prefix) {
+				parts.push('[', this.prefix, '] ')
```

Comment:
In the same way as the comment below, it would be nice if the values for `prefix` were constants (`PREFIX_FROM`, `PREFIX_ON`,...) and the translation happened here.